homeassistant/components/livisi/climate.py

```diff
@@ -145,10 +149,25 @@ async def async_added_to_hass(self) -> None:
     def set_hvac_mode(self, hvac_mode: HVACMode) -> None:
         """Do nothing as LIVISI devices do not support changing the hvac mode."""
 
+    def _update_hvac_state(self) -> None:
+        """Calculate current hvac state based on target and current temperature."""
+        if (
+            self._attr_current_temperature is None
+            or self._attr_target_temperature is None
+        ):
+            self._attr_hvac_action = HVACAction.OFF
+        elif self._attr_target_temperature > self._attr_current_temperature:
+            self._attr_hvac_action = HVACAction.HEATING
+        elif self._attr_target_temperature == self._attr_min_temp:
+            self._attr_hvac_action = HVACAction.OFF
+        else:
+            self._attr_hvac_action = HVACAction.IDLE
```

Comment:
So, the device actually doesn't support this, so we are assuming? If that is the case, I suggest not implementing this. We shouldn't emulate/assume.

Reply:
This was also my inital understanding - see [my comment here](https://github.com/home-assistant/core/issues/90630#issuecomment-1497247956) - but as the bug was reopened and I understand the rationale behind the request (showing the state in diagrams correctly) I now think this makes sense.

If you don't agree, please close the issue and the PR.